File: src/event/impl/server/ServerCrashEvent.php
<?php

namespace pocketcloud\cloud\event\impl\server;

use pocketcloud\cloud\server\CloudServer;

class ServerCrashEvent extends ServerEvent {

    public function __construct(
        CloudServer $server,
        private readonly array $data
    ) {
        parent::__construct($server);
    }

    public function getData(): array {
        return $this->data;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/event/impl/server/ServerCrashEvent.java
package de.pocketcloud.cloud.event.impl.server;

import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;

import java.util.List;

public class ServerCrashEvent extends ServerEvent {

    @Getter
    private final List<String> data;

    public ServerCrashEvent(CloudServer server, List<String> data) {
        super(server);
        this.data = data;
    }
}
